src/transactions/TransactionFrame.cpp

```diff
@@ -59,25 +60,60 @@ int64_t const MAX_RESOURCE_FEE = 1LL << 50;
 using namespace std;
 using namespace stellar::txbridge;
 
-TransactionFrame::TransactionFrame(Hash const& networkID,
-                                   TransactionEnvelope const& envelope)
-    : mEnvelope(envelope), mNetworkID(networkID)
+static bool
+hasDexOperationsInternal(TransactionResultPayload const& resPayload)
 {
-    // Create operation frames with dummy results. Currently the proper results
-    // are initialized in `TransactionFrame::resetResults` and eventually the
-    // operation frames should be decoupled from the results completely and
-    // created just once.
-    auto& ops = mEnvelope.type() == ENVELOPE_TYPE_TX_V0
-                    ? mEnvelope.v0().tx.operations
-                    : mEnvelope.v1().tx.operations;
-    getResult().result.code(txFAILED);
-    getResult().result.results().resize(static_cast<uint32_t>(ops.size()));
+    for (auto const& op : resPayload.getOpFrames())
+    {
+        if (op->isDexOperation())
+        {
+            return true;
+        }
+    }
+    return false;
+}
 
-    for (size_t i = 0; i < ops.size(); i++)
+static bool
+isSorobanInternal(TransactionResultPayload const& resPayload)
```

Comment:
Using the whole result as an argument here looks a bit weird (I've also left a comment re operation frames in another file). Can we just accept ops frames for now in this and the other methods that only access operations? That would at least look more logical.